scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -554,3 +555,99 @@ def test_converged(self):
         solver = DifferentialEvolutionSolver(rosen, [(0, 2), (0, 2)])
         solver.solve()
         assert_(solver.converged())
+
+    def test_constraint_setup(self):
```

Comment:
this unit test is pretty long -- might benefit from breaking into discrete tests, although not critical I suppose

Reply:
All the tests in there are looking at various numeric combinations for examining `_constraint_violation_fn` or `_calculate_population_feasibilities`; they're logically grouped, it's just that there are several of them.

Reply:
I understand that they're logically grouped, but at some point grouping of several smaller functions within a test class is warranted to improve reporting of failures